auth/providers/azure/pop_tokenverifier_test.go

```diff
@@ -143,160 +144,330 @@ const (
 	headerBadtypMissing = "headerBadtypMissing"
 	uClaimsMissing      = "uClaimsMissing"
 	tsClaimsMissing     = "tsClaimsMissing"
+	atClaimsMissing     = "atClaimsMissing"
+	atClaimIncorrect    = "atClaimIncorrect"
 	cnfClaimsMissing    = "cnfClaimsMissing"
+	cnfJwkClaimsEmpty   = "cnfJwkClaimsEmpty"
 	cnfJwkClaimsWrong   = "cnfJwkClaimsWrong"
+	cnfJwkClaimsMissing = "cnfJwkClaimsMissing"
+	accessTokenCnfWrong = "accessTokenCnfWrong"
+	atClaimsWrongType   = "atClaimsWrongType"
 	atCnfClaimMissing   = "atCnfClaimMissing"
 	atCnfClaimWrong     = "atCnfClaimWrong"
+	tsClaimsTypeString  = "tsClaimsTypeString"
+	tsClaimsTypeUnknown = "tsClaimsTypeUnknown"
+	uClaimsWrongType    = "uClaimsWrongType"
+	signatureWrongType  = "signatureWrongType"
 )
 
-func GeneratePoPToken(ts int64, hostName, kid string) (string, error) {
-	if strings.Contains(kid, badTokenKey) {
-		return kid, nil
+// A struct that represents a PoP token
+type PoPToken struct {
```

Comment:
Can we move all these helper struct / function into a dedicated file (`pop_tokenverifier_utils_test.go` ?) and just have the `TestPopTokenVerifier_Verify(t *testing.T)` func in this one ?